### PR TITLE
make extension restartless + add new prefs for Oculus 0.8 runtime & Nightly (fixes #20; fixes #22) + disable e10s (fixes #15)

### LINKED ISSUE #15: e10s message needs to be disabled
We should suppress e10s or customize the message to warn WebVR users that turning on e10s will stop WebVR from working.

image

At least until we get 1182048 "Make webvr and e10s work together" resolved:
https://bugzilla.mozilla.org/show_bug.cgi?id=1182048

### LINKED ISSUE #20: Update add-on to set browser prefs with newest 0.8 runtime
See post by @kearwood at:
https://www.reddit.com/r/WebVR/comments/3zmclh/webvr_enabled_by_default_in_firefox_nightly/
- Set the layout.frame_rate preference (in about:config) to 75
- Un-check the "Enable Multi-process Nightly" option in Preferences.

### LINKED ISSUE #22: Add-on should toggle vr.enabled to default when uninstalling
When uninstalling, the vr.enabled pref is set to false.

### PR DESCRIPTION
**Pull request by cvan**
testing welcome!

### CODE REVIEW

**Comment by cvan** [comment]
this also handles #15

**Comment by cvan** [comment]
extension: https://drive.google.com/file/d/0B52bDqlwf1F9UUF2TTVfeUljZjQ/view?usp=sharing

**Comment by cvan** [comment]
using the [aforementioned `.xpi` extension I've built](https://drive.google.com/file/d/0B52bDqlwf1F9UUF2TTVfeUljZjQ/view?usp=sharing), I was able to confirm that everything works great :tada:.

I tested with the following browsers:
1. Firefox release channel - 43.0.4
2. Firefox Developer Edition - 46.0a2 (2016-01-28)
3. Firefox Nightly - 47.0a1 (2016-01-28)

I tested with the following platforms:
1. Windows 8
2. Mac OS X El Capitan

I tested the following scenarios:
1. Upgrade 0.2.0 extension to 0.3.0 extension → VR still works, prefs correctly set (and new ones added)
   1. Uninstall 0.3.0 extension → restart Firefox → prefs correctly reverted (and new ones removed)
2. Install 0.3.0 extension from scratch -> VR immediately works (**without a browser restart**), prefs correctly set
   1. Uninstall 0.3.0 extension → restart Firefox → prefs correctly reverted (and new ones removed)

<hr>

would be great to get a couple more folks to test. @caseyyee? @dmarcos? anyone with immediate access to Windows machines …

File: bootstrap.js [hunk 1]
```diff
@@ -4,42 +4,49 @@
 
 Components.utils.import("resource://gre/modules/Services.jsm");
 
-function startup(data, reason)
-{
-  Services.prefs.setBoolPref("dom.vr.enabled", true);
+const PREFS = {
+  // Enables WebVR. Not needed for Nightly, but we need to keep for other versions.
+  "dom.vr.enabled": true,
```
Comment (caseyyee) [inline bootstrap.js]: `gfx.vr.mirror-textures = true`
see: https://bugzilla.mozilla.org/show_bug.cgi?id=1196435

`dom.vr.add-test-devices = 1`
If we don't set this, VR does not work unless HMD is plugged in.   

I feel that these should both be set, but I know @kearwood may disagree (and reasons for this)

Comment (cvan) [inline bootstrap.js]: ping @kearwood?

Comment (kearwood) [inline bootstrap.js]: I highly recommend against setting dom.vr.add-test-devices to 1, as this effectively disables the only way for web authors to detect that you have any VR hardware.

Comment (kearwood) [inline bootstrap.js]: As per gfx.vr.mirror-textures, the reason not to set to "True" would be performance.  Additionally, we don't want content to expect that mirroring is assumed as the upcoming API changes expect mirroring to be done by the rendering engine (three.js, aframe, etc)

**Comment by kearwood** [comment]
I recommend that we set these preferences in the updated addon:

dom.vr.enabled = true
(Enables WebVR, no needed for nightly, but need to keep for other versions)

browser.tabs.remote.autostart.2 = false
(Disables e10s)

dom.vr.poseprediction.enabled = true
(Enables pose prediction.  This pref is not yet added to nightly, but
will be landed early next week.  I think its safe for you to add it in
advance)

**Comment by cvan** [comment]
@kearwood all of those were already set in the extension

**Comment by cvan** [comment]
@caseyyee @kearwood: could y'all take a look at my recent updates here? I simplified things a ton and did a lot of testing.

I also have some branches with cool things that introduce simple add-on prefs for toggling mirroring. the hope is to soon/eventually introduce some of the functionality @caseyyee was working in originally in https://github.com/MozVR/webvr-addon-v2.

who knew add-ons could be so fun!?

File: data/content-script.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+self.port.on('msg', function (msg) {
+  msg = parseMsg(msg);
+  if (!msg) { return; }
+
+  var type = msg.type;
+  var data = msg.data;
+
+  switch (type) {
+    case 'version':
+      WEBVRPLUS.version = data;
```
Comment (cvan) [inline data/content-script.js:33]: this lets us sniff on the client:

``` shell
>>> window.WEBVRPLUS
{version: "0.3.0"}
```

Comment (cvan) [inline data/content-script.js:33]: I have more stuff I want to add but am starting here

**Comment by cvan** [comment]
latest version to test: https://drive.google.com/file/d/0B52bDqlwf1F9V04yN2ZyZkNtME0/view?usp=sharing

File: bootstrap.js (hunk 1 above)
Comment (cvan) [inline bootstrap.js]: I will enable this as a user option for this extension in a later PR

**Comment by cvan** [comment]
a few hours ago I submitted to [AMO](https://addons.mozilla.org) this as a new version of the extension. if anyone finds anything, let me know ASAP. things look good to me.

**Pull request merged**